data/campaigns/Under_the_Burning_Suns/utils/abilities.cfg

```diff
@@ -1,5 +1,433 @@
+        active_on = offense
+    [/dummy]
+#enddef
+
+#define WEAPON_SPECIAL_NOVA_EVENTS
+    [event]
+        name = attacker_hits
+        id = nova
+        first_time_only = no
+        [filter_attack]
+            special_id = nova
+        [/filter_attack]
+        [harm_unit]
+            [filter]
+                [not]
+                    find_in = second_unit
+                [/not]
+                [filter_adjacent]
+                    find_in = unit
+                [/filter_adjacent]
+            [/filter]
+            amount = $weapon.damage
+            damage_type = $weapon.type
+            alignment = $unit.alignment
+            kill = yes
+            experience = yes
+            fire_event = yes
+        [/harm_unit]
+    [/event]
+#enddef
+
+#define WEAPON_SPECIAL_RAY
+    [dummy]
+        id = ray
+        name = _"ray"
+        description= _ "This weapon, when used offensively, also deals damage to a unit behind the target."
+        active_on = offense
+    [/dummy]
+#enddef
+
+#define WEAPON_SPECIAL_RAY_EVENTS
+    [event]
+        name = attacker_hits
+        id = ray_hits
+        first_time_only = no
+        [filter_attack]
+            special_id = ray
+        [/filter_attack]
+        [harm_unit]
+            [filter]
+                [filter_adjacent]
+                    find_in = second_unit
+                    formula = "
+                        unit_at(direction_from(self.loc,caster.facing)) = other
+                    where
+                        caster = unit_at(direction_from(self.loc, self.facing))
+                    "
+                [/filter_adjacent]
+            [/filter]
+            amount = $weapon.damage
+            damage_type = $weapon.type
+            alignment = $unit.alignment
+            kill = yes
+            experience = yes
+            fire_event = yes
+        [/harm_unit]
+    [/event]
+#enddef
+
+#define UTBS_ABILITY_TAILWIND X
+    [tailwind]
+        id = tailwind
+        value = {X}
+    [/tailwind]
+#enddef
+
+#define UTBS_TAILWIND_EVENTS
+    [event]
+        name=turn refresh
+        first_time_only=no
+        id=tailwind
+
+        # This hardcodes that only tailwind+1 and tailwind+2 are supported, which simplifies the code to avoid them
+        # stacking (so a unit adjacent to both a tailwind+1 and a tailwind+2 will only get +2 rather than +3).
+        # If more levels should be supported, a loop can be added, but that will make the code more complex.
+        [store_unit]
+            [filter]
+                ability=tailwind
+                [filter_side]
+                    [allied_with]
+                        side=$side_number
+                    [/allied_with]
+                [/filter_side]
+                [and]
+                    # filter_wml is slow, which is why there's already an ability=tailwind filter before this
+                    [filter_wml]
+                        [abilities]
+                            [dummy]
```

Comment:
The `dummy` tags in the `filter_wml` need to change too (and there's another on line 379).

Reply:
Sorry, thanks and done.